WordPressKit/MultipartForm.swift

```diff
@@ -0,0 +1,162 @@
+import Foundation
+
+enum MultipartFormError: Swift.Error {
+    case inaccessbileFile(path: String)
+    case impossible
+}
+
+struct MultipartFormField {
+    let name: String
+    let filename: String?
+    let mimeType: String?
+    let bytes: UInt64
+
+    fileprivate let inputStream: InputStream
+
+    init(text: String, name: String, filename: String? = nil, mimeType: String? = nil) {
+        self.init(data: text.data(using: .utf8)!, name: name, filename: filename, mimeType: mimeType)
+    }
+
+    init(data: Data, name: String, filename: String? = nil, mimeType: String? = nil) {
+        self.inputStream = InputStream(data: data)
+        self.name = name
+        self.filename = filename
+        self.bytes = UInt64(data.count)
+        self.mimeType = mimeType
+    }
+
+    init(fileAtPath path: String, name: String, filename: String? = nil, mimeType: String? = nil) throws {
```

Comment:
Have you considered removing `filename` and forcing callers to specify the whole path with `fileAtPath`?

Reply:
These two things are related but also should co-exist. The multipart form contains a `filename` field, whose value will be set by the `filename` argument here. We can use the file name of the  `fileAtPath` as the default value. But if some code passes a `filename` argument in, we should use it instead.